app/models/assignment_invitation.rb

```diff
@@ -0,0 +1,50 @@
+class AssignmentInvitation < ActiveRecord::Base
+  belongs_to :assignment
+
+  validates_presence_of   :key
+  validates_uniqueness_of :key
+
+  after_initialize :assign_key
+
+  def redeem(user)
```

Comment:
This seems like a lot for one method, I'll probably break this up into a few smaller parts. @johndbritton do you think this should stay on the model, or should these become service objects?